lib/Hooks.js

```diff
@@ -1,76 +1,76 @@
-const hooksLog = require('./hooksLog');
+import hooksLog from './hooksLog'
 
 // READ THIS! Disclaimer:
 // Do not add any functionality to this class unless you want to expose it to the Hooks API.
 // This class is only an interface for users of Dredd hooks.
 
 class Hooks {
   constructor(options = {}) {
-    this.before = this.before.bind(this);
-    this.beforeValidation = this.beforeValidation.bind(this);
-    this.after = this.after.bind(this);
-    this.beforeAll = this.beforeAll.bind(this);
-    this.afterAll = this.afterAll.bind(this);
-    this.beforeEach = this.beforeEach.bind(this);
-    this.beforeEachValidation = this.beforeEachValidation.bind(this);
-    this.afterEach = this.afterEach.bind(this);
-    this.log = this.log.bind(this);
-    ({ logs: this.logs, logger: this.logger } = options);
-    this.beforeHooks = {};
-    this.beforeValidationHooks = {};
-    this.afterHooks = {};
-    this.beforeAllHooks = [];
-    this.afterAllHooks = [];
-    this.beforeEachHooks = [];
-    this.beforeEachValidationHooks = [];
-    this.afterEachHooks = [];
+    this.before = this.before.bind(this)
+    this.beforeValidation = this.beforeValidation.bind(this)
+    this.after = this.after.bind(this)
+    this.beforeAll = this.beforeAll.bind(this)
+    this.afterAll = this.afterAll.bind(this)
+    this.beforeEach = this.beforeEach.bind(this)
+    this.beforeEachValidation = this.beforeEachValidation.bind(this)
+    this.afterEach = this.afterEach.bind(this)
+    this.log = this.log.bind(this)
+    ;({ logs: this.logs, logger: this.logger } = options)
```

Comment:
Doesn't seem like the semi-colons are removed in the other files I've reviewed so far, why are they removed from other lines in this file and inserted at the start of this line instead of the end of the prior line?

Reply:
1. I will align semicolons consistency and remove them from the source code altogether.
2. Because `({ logs: this.logs, logger: this.logger } = options)` is not a valid expression per formatting tools, so it appends an explicit semicolon in no-semi mode to let it be evaluated properly (and not the part of the previous expression).

Reply:
That's black sorcery. Can we replace all expressions starting with a semicolon with some boring JavaScript code?

```js
this.logs = options.logs
this.logger = options.logger
```

Reply:
Yes, sure. Moreover, we can replace them in the original code as one of the first PRs:

https://github.com/apiaryio/dredd/blob/91ceef1a1b74191761cafc5d11ccb17d4d3b1e75/lib/Hooks.js#L18